### DISCUSSION TITLE
Intendation of right curly bracket in code formatting is not logical?

### DISCUSSION THREAD

**Discussion by jbrepogmailcom** [Cyfrin/foundry-full-course-cu#272] [General]
Hello. I think that when you close block of code that is within curly brackets, the right curly bracket, that closes the code block, actually belongs to that block and not to block which is one level up. Therefore that right curly bracket should be intended at same level as the block code that it closes. However, in all codes it is intended at same level as the upper owning block.

Is there any settings in vscodium solidity automatic formatter, that formats right curly bracket exactly below the code block it closes, not one step to the left?
https://en.wikipedia.org/wiki/Indentation_style#Ratliff_style

**Comment by ChefAharoni** [comment]
If I understand correctly what you're asking, then yes - there is an auto-formatter called **[Prettier](https://prettier.io/)** that automatically formats your code. 
You can also set in VSCode settings that it will auto format when saving the document - saves a lot of time for me.

<img width="1000" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/4399057/ce43444f-3ed8-4898-a44b-37f7df8e3574">
